src/cozmo/action.py

```diff
@@ -65,6 +74,212 @@
 _VALID_STATES = {ACTION_IDLE, ACTION_RUNNING, ACTION_SUCCEEDED, ACTION_FAILED, ACTION_ABORTING}
 
 
+class _ActionResult(namedtuple('_ActionResult', 'name id')):
+    # Tuple mapping between CLAD ActionResult name and ID
+    # All instances will be members of ActionResults
+
+    # Keep _ActionResult as lightweight as a normal namedtuple
+    __slots__ = ()
+
+    def __str__(self):
+        return 'ActionResults.%s' % self.name
+
+
+class ActionResults:
+    """The possible result values for an Action.
+
+    An Action's result is set when the action completes.
+    """
+
+    #: Action completed successfully.
+    SUCCESS = _ActionResult("SUCCESS", _clad_to_game_cozmo.ActionResult.SUCCESS)
+
+    #: Action is still running.
+    RUNNING = _ActionResult("RUNNING", _clad_to_game_cozmo.ActionResult.RUNNING)
+
+    #: Action was cancelled (e.g. via :meth:`~cozmo.robot.Robot.abort_all_actions` or
+    #: :meth:`Action.abort`).
+    CANCELLED = _ActionResult("CANCELLED", _clad_to_game_cozmo.ActionResult.CANCELLED)
+
+    #: Action aborted itself (e.g. had invalid attributes, or a runtime failure).
+    ABORT = _ActionResult("ABORT", _clad_to_game_cozmo.ActionResult.ABORT)
+
+    #: Animation Action aborted itself (e.g. there was an error playing the animation).
+    ANIM_ABORTED = _ActionResult("ANIM_ABORTED", _clad_to_game_cozmo.ActionResult.ANIM_ABORTED)
+
+    #: There was an error related to vision markers.
+    BAD_MARKER = _ActionResult("BAD_MARKER", _clad_to_game_cozmo.ActionResult.BAD_MARKER)
+
+    #: There was a problem with the Object ID provided (e.g. there is no Object with that ID).
+    BAD_OBJECT = _ActionResult("BAD_OBJECT", _clad_to_game_cozmo.ActionResult.BAD_OBJECT)
+
+    #: There was a problem with the Pose provided.
+    BAD_POSE = _ActionResult("BAD_POSE", _clad_to_game_cozmo.ActionResult.BAD_POSE)
+
+    # (Undocumented) The SDK provided tag was bad (shouldn't occur - would indicate a bug in the SDK)
```

Comment:
A number of these results only make sense for internal dev (or factory test/calibration), so I felt it best to keep them out of the API documentation to avoid confusion. The thing that's keeping them out of the API docs is just that it's commented with just a hash `#` instead of a hash-colon `#:` so it won't get a docstring - I just added the "(Undocumented)" text to help clarify that it was deliberately excluded. I guess it's not that clear on reflection...